scripts/run-tests

```diff
@@ -0,0 +1,95 @@
+#! /usr/bin/env bash
+#
+# Validate rules and run semgrep to check that it's producing the expected
+# results.
+#
+# This was originally part of the makefile but it's gotten too complicated
+# due the need for extra escaping and error codes being ignored.
+#
+set -eu
+
+usage() {
+  cat <<EOF
+Usage: $0 [validate|test]
+
+validate: check only rule validity
+test: run semgrep rules against the target files
+
+Return non-zero if some validation or test doesn't meet the expectations.
+EOF
+}
+
+error() {
+  echo "Error: $*" >&2
+  exit 1
+}
+
+# Use the SEMGREP environment variable to specify a non-standard semgrep
+# command. This is useful for calling a development version of semgrep
+# e.g.
+#   PIPENV_PIPFILE=~/semgrep/cli/Pipfile SEMGREP='pipenv run semgrep' make test
+#
+if [[ -z "$SEMGREP" ]]; then
```

Comment:
@mjambon does this script work?
I've just tried make test in semgrep-rules and got:
```
(cli) [pad@thinkstation semgrep-rules (develop)]$ make test
make validate
make[1]: Entering directory '/home/pad/github/semgrep-rules'
./scripts/run-tests validate
./scripts/run-tests: line 32: SEMGREP: unbound variable
make[1]: *** [Makefile:13: validate] Error 1
make[1]: Leaving directory '/home/pad/github/semgrep-rules'
make: *** [Makefile:8: test] Error 2
```